Zurb Foundation 5 Admin Template

ondoheer opened the pull request:
Hi, as promised, I have finished a basic zurb foundation 5 template.
I have left many of the classes that were used, since I never use BS, when I wasn't sure if they were BS classes or utility ones I just left them. This can be polished.

- I have opinionatedly decided to use a fixed top menu bar
- I have ised svg iconfonts (foundation ones) and haven't set a png fallback because of how Foundation handles it, it's not just a class, it's a diferent tag. Since Foundation 5 is supposed to aim at the latest browsers I thought this seemed coherent. Still we can work a way to check a variable with modernizer and chage it.
- I have added an app.css file with little style changes I had to force for everything to look nice.
- I have tried to test the template with all the examples, but I wasn't able to do so with some of them.
- It has some problems on mobile, but they are the sam BS3 is having, it would need to reformat how the filters and their forms are being produced.
- I need to find a way to control the width of the columns on the tables without hacking them. 
- Forms and Tables don't need clases to be Foundation styled.

I hope this helps the community, I can keep on collaboarting with this, I will use it on a real project at our office as soon as it's merged, so I will be able to test it properly and add suggestions.

Have a nice wek!

rochacbruno commented:
:+1: Nice! waiting for this, I will do some tests

klinkin commented:
:+1:

pawl-rs commented on examples/zurb5/app.py:
```diff
@@ -0,0 +1,151 @@
+
+# Customized admin interface
+class CustomView(ModelView):
+    list_template = 'list.html'
```
These files don't exist in the examples/zurb5/ folder.

ondoheer replied:
Yes, this happens because the examples asume Flask-Admin is installed on your system. Until the zurb5 templates is included in the package, it won't be updated in the package installed with pip. You would have to take the eamplefile to the root of yur project and put this packaage there as well, then import it from there instead of the installed one.

pawl-rs replied:
I checked it out from your fork and used ```python setup.py install```.

This code assumes there following files exist:
* examples/zurb5/templates/create.html
* examples/zurb5/templates/edit.html
* examples/zurb5/templates/list.html

To get this example to work, you can just remove ```class CustomView(ModelView):```.

ondoheer replied:
You are right, I will correct this.

ondoheer replied:
Done.

pawl commented:
Can you also add support for modals?

Also looks like this is going to break column_editable_list (http://flask-admin.readthedocs.org/en/latest/api/mod_model/#flask_admin.model.BaseModelView.column_editable_list): https://github.com/vitalets/x-editable/issues/10

ondoheer commented:
Hi Paul, will check that as soon as I can. To help me do it faster, could you point to me an example where modals are boing used with Bootstrap?

pawl commented:
Here are some examples:

Modal macros in lib.html to create buttons and the modal window:
https://github.com/flask-admin/flask-admin/blob/master/flask_admin/templates/bootstrap3/admin/lib.html#L100

Adding modal window to list.html:
https://github.com/flask-admin/flask-admin/blob/master/flask_admin/templates/bootstrap3/admin/model/list.html#L175

Check if modal is enabled and change presentation in edit.html and create.html:
https://github.com/flask-admin/flask-admin/blob/master/flask_admin/templates/bootstrap3/admin/model/edit.html#L18
https://github.com/flask-admin/flask-admin/blob/master/flask_admin/templates/bootstrap3/admin/model/create.html#L18

Adding buttons to list.html: https://github.com/flask-admin/flask-admin/blob/master/flask_admin/templates/bootstrap3/admin/model/list.html#L22

I think you're going to do most of it with javascript since Foundation doesn't load stuff from data-remote by default like Bootstrap does: http://foundation.zurb.com/forum/posts/1399-adding-reveal-links-via-javascriptajax

ondoheer commented:
Thanks Paul,
will check this after the weekend. Have to run some tests. Do I make another PR once this is solved?

petrus-jvrensburg commented:
@ondoheer, no need for another PR. We can keep this open as long as you need.

ondoheer commented:
Hello Petrus,

got that. I have run into some problems.

1) I haven't managed to get Flask-admin  1.2.0 to show me modals for editing nor creating when using BS2 or BS3, I tried modifying the examples and then a personal project of mine. It's like it's not reading the properties and the template is just showing me the regular links (no modal=true on the URI).

2) The modals are being created on my theme, but the content comes out empty, I wanted to know how you guys are handling it.

3) I'm still working on getting x-editable to work with plain jQuery or jQuery-UI

petrus-jvrensburg commented:
@ondoheer The modal window feature is brand new, so it's not available on the latest pypi build. To try it out, you'll have to install the latest version straight from the github repo. You can do this with pip:
```
pip install git+https://github.com/flask-admin/flask-admin.git@master
```

Then, if you `pip freeze` you should see:
```
Flask-Admin==1.2.1-dev
```
Hope that helps.

ondoheer commented:
Sorry, I didin't realize this master wasn't synced with PyPi, that explains a few "bugs" I encountered! Thanks for that.

ondoheer commented:
Hello guys,

just found some more time to work on the project today, I have stomped against some problems:

1. I'm using x-editable jqueryui version, but it seems to run into [Bug 687 ](https://github.com/vitalets/x-editable/issues/687). I updated the js as stated in [PR 724](https://github.com/vitalets/x-editable/pull/724) but that just managed to silence the JS error.

2. I tried to use pure jquery x-editable but that just caused more problems. It seems the problem has to do with setting x-editable to popup mode. I tried to find where to change that to inline mode in the flask.admin code, but didin't find it. Before just overriding anytrhing I wanted to consult with you where I should do this and how. 

3. I have updated the modal templates, they now work *foundation style*, but they don't extend the templates with models/modal/{create, edit}.html as they should. Haven't found why yet. Any help would be appreciated.

4. I have tested the modals with layout from BS2, they appear and render the form, but they aren't working when trying to send the data. I have been working over BS2 templates because I always read  the project managers don't feel BS3 is ready to overtake BS as the default theme.

Any help getting this to work would be awesome, I'm, kind of swamped at the office and would like to help this get through as best as I can.

m1guelpf commented:
Any updates?

ondoheer commented:
Hello Miguel,

for personal reasons I couldn't finish this, we were doing it so it became
one of the comany tools.

On Sun, Aug 6, 2017 at 6:51 AM, Miguel Piedrafita <notifications@github.com>
wrote:

> Any updates?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/flask-admin/flask-admin/pull/937#issuecomment-320502161>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADgsFBvJZIXn4DIPfIowHUp5tAeGDmO2ks5sVajNgaJpZM4FUn0m>
> .
>



-- 




*P E D R O  B A U M A N N  C O R N E J O*
PSICÓLOGO - PSICOTERAPEUTA - FLORES DE BACH
PROGRAMADOR BACKEND - ATHELAS PERÚ
LIMA, PERÚ
